src/Components/Web.JS/src/Platform/WebAssemblyProgressReporter.ts

```diff
@@ -0,0 +1,19 @@
+export class WebAssemblyProgressReporter {
+    /**
+    * Modifies CSS of the loading ring to reflect linear progress based on the number of resources.
+    * @param resourcesTotal The total number of resources retrieved from {@link ./Mono/MonoPlatform.ts} class.
+    * @param resourcesLoaded The current number of resources loaded retrieved from {@link ./WebAssemblyResourceLoader.ts} class.
+    */
+    static setProgress(resourcesTotal: number, resourcesLoaded: number): void {
+        const circle = document.getElementById('blazor-default-loading-progress') as unknown as SVGCircleElement;
+        const circumference = 2 * Math.PI * circle.r.baseVal.value;
+        const progressPercentage = resourcesLoaded / resourcesTotal;
+        const ring = (1 - progressPercentage) * circumference;
+        circle.style.strokeDasharray = `${circumference} ${circumference}`;
+        circle.style.strokeDashoffset = `${ring}`;
+        circle.style.display = 'block';
+        const element = document.getElementById('blazor-default-loading-percentage') as unknown as SVGTextElement;
+        const displayPercentage = Math.floor(progressPercentage * 100);
+        element!.textContent = `${displayPercentage}%`;
+    }
+}
```

Comment:
Shouldn't this be on the template? Otherwise, we are coupling the framework with the default implementation on the template, aren't we?

Reply:
> we are coupling the framework with the default implementation on the template, aren't we?

Granted this does introduce some coupling between the framework and template, however this is in line with other default handlers we have within Blazor, like the [DefaultReconnectDisplay](https://github.com/dotnet/aspnetcore/blob/main/src/Components/Web.JS/src/Platform/Circuits/DefaultReconnectDisplay.ts). By keeping this code in the framework, we can minimize the amount of JS within `index.html` (or eliminate the need to have a separate `wwwroot/js/loading.js` file). I feel like this degree of coupling isn't the worst thing given this key benefit of template simplicity. 

In fact, we may even want to simplify the additions to `index.html`, to create the `svg` element using JS (assuming that's possible) so that the only addition to `index.html` is:

```diff
-     <div id="app">Loading...</div>
+    <div id="app">
+        <div id="blazor-default-loading"></div>
+    </div>
```